### DISCUSSION TITLE
Fork and Pull Request Workflow

### DISCUSSION THREAD

**Discussion by xanjohns** [BetaHuhn/repo-file-sync-action#133] [Ideas] [answered]
Hello!

I'm currently working with [SymbiFlow](https://symbiflow.github.io/), an open-source organization on Github, and we are looking for a way to keep files consistent across repositories, similar to what your action here accomplishes. However, we use a fork and pull request workflow and do not want bots or actions having direct access to modify upstream repositories.

What do you think of adding a feature that would allow a fork and pull request workflow using this action? 

A possible solution could be something like the following:
- Add a `FORK` parameter that specifies the bot/user account for the forks
- Clone the upstream repository in `sync.yml` as normal, but push new commits to bot's fork
- Open pull request from fork to upstream repository

Checking whether the appropriate forks already exist/creating new ones could also be included, or done by another action if it would be easier.

I'd be happy to contribute if this seems like a feature that could be added.

**Comment by BetaHuhn** [comment]
Hey, thanks for taking the time to open this discussion!

I myself have no need for such a feature, but I can see the value it provides.

That being said, I don't want this action to become too complicated for me to maintain and for others to use, so it depends on the implementation/scope of the feature.

How do you imagine the `FORK` parameter to work? Would it be global for all target repositories specified in the `sync.yml` or on an individual basis?

**Reply by xanjohns** [reply]
I have opened up a [pull request](https://github.com/BetaHuhn/repo-file-sync-action/pull/135) with a possible solution for how I could see this implemented.

The way I imagined it, the `FORK` parameter would be global and only specified once as an input to the action file.

If this is something you would like merged in and have suggestions on how it could be implemented better, please let me know! If this is beyond the vision/goals of the action and it would be better not to merge it in, that would be fine too.

Thanks!

**Reply by BetaHuhn** [reply]
Thanks for creating a PR, will take a closer look next week!

The changes don't look too complicated and I can see myself merging this.

**Reply by BetaHuhn** [reply]
Merged, thanks again @xanjohns!

**Comment by BetaHuhn** [comment, marked as answer]
:tada: This feature was added in version [v1.16.0](https://github.com/BetaHuhn/repo-file-sync-action/releases/tag/v1.16.0) :tada:

To use the latest version, use the Action with the `v1` tag:

```yml
uses: BetaHuhn/repo-file-sync-action@v1
```

Or the `latest` tag:

```yml
uses: BetaHuhn/repo-file-sync-action@latest
```
